src/System.Linq.Expressions/src/System/Linq/Expressions/NewExpression.cs

```diff
@@ -183,6 +183,8 @@ public static NewExpression New(ConstructorInfo constructor, IEnumerable<Express
         public static NewExpression New(ConstructorInfo constructor, IEnumerable<Expression> arguments, IEnumerable<MemberInfo> members)
         {
             ContractUtils.RequiresNotNull(constructor, nameof(constructor));
+            ContractUtils.RequiresNotNull(constructor.DeclaringType, nameof(constructor) + "." + nameof(constructor.DeclaringType));
```

Comment:
Is this reachable? Can DeclaringType be null?

Also not clear, even if possible, should it be an argument exception? 
It would not look like a direct fault of the caller. Perhaps it should just NRE if such thing happened.